industrial_extrinsic_cal/src/ros_camera_observer.cpp

```diff
@@ -612,23 +617,23 @@ void ROSCameraObserver::triggerCamera()
     if(loaded_color_image.data && loaded_mono_image.data){
       ROS_DEBUG("Loaded it");
     }
+    image_number_++;
   }
   else{
     ROS_DEBUG("rosCameraObserver, waiting for image from topic %s",image_topic_.c_str());
     bool done=false;
     while(!done){
-      sensor_msgs::ImageConstPtr recent_image = ros::topic::waitForMessage<sensor_msgs::Image>(image_topic_);
+      sensor_msgs::ImageConstPtr recent_image = ros::topic::waitForMessage<sensor_msgs::Image>(image_topic_, ros::Duration(5.0));
+      if (! recent_image)
+      {
+        ROS_ERROR("No image acquired on topic '%s'", image_topic_.c_str());
+        break;
+      }
       
       ROS_DEBUG("captured image in trigger");
       try
       {
-        if(recent_image->encoding == "mono16"){  // asus and kinect ir images are mono16, bridge mishandles conversion to mono8
-          input_bridge_ = cv_bridge::toCvCopy(recent_image, "mono16");
-          input_bridge_->image.convertTo(input_bridge_->image, CV_8UC1, 1.0, 0.0);
-        }
-        else{
-          input_bridge_ = cv_bridge::toCvCopy(recent_image, "mono8");
-        }
+        input_bridge_ = cv_bridge::toCvCopy(recent_image, "mono8");
```

Comment:
Forgot to bring attention to this change as well. I found the manual conversion from `mono16` to `mono8` wasn't doing proper scaling and needed a different factor in `convertTo`. I tried the `cv_bridge` conversion and it works fine so I just removed this special case.